src/test/java/io/nats/client/PublishTests.java

```diff
@@ -261,4 +263,74 @@ public void testMaxPayload() throws Exception {
         }
         catch (IllegalArgumentException ignore) {}
     }
+
+    @Test
+    public void testUtf8Subjects() throws Exception {
+        String subject = dataAsLines("utf8-test-strings.txt").get(0);
+        String jsSubject = variant() + "-" + subject; // just to have a different;
+
+        AtomicReference<String> coreReceivedSubjectNotSupported = new AtomicReference<>();
+        AtomicReference<String> coreReceivedSubjectWhenSupported = new AtomicReference<>();
+        AtomicReference<String> jsReceivedSubjectNotSupported = new AtomicReference<>();
+        AtomicReference<String> jsReceivedSubjectWhenSupported = new AtomicReference<>();
+        CountDownLatch coreReceivedLatchNotSupported = new CountDownLatch(1);
+        CountDownLatch coreReceivedLatchWhenSupported = new CountDownLatch(1);
+        CountDownLatch jsReceivedLatchNotSupported = new CountDownLatch(1);
+        CountDownLatch jsReceivedLatchWhenSupported = new CountDownLatch(1);
+
+        try (NatsTestServer ts = new NatsTestServer(false, true);
+             Connection ncNotSupported = standardConnection(standardOptionsBuilder(ts.getURI()).build());
+             Connection ncSupported = standardConnection(standardOptionsBuilder(ts.getURI()).supportUTF8Subjects().build()))
+        {
+            try {
+                ncNotSupported.jetStreamManagement().addStream(
+                    StreamConfiguration.builder()
+                        .name(stream())
+                        .subjects(jsSubject)
+                        .build());
+                JetStream jsNotSupported = ncNotSupported.jetStream();
+                JetStream jsSupported = ncNotSupported.jetStream();
+
+                Dispatcher dNotSupported = ncNotSupported.createDispatcher();
+                Dispatcher dSupported = ncSupported.createDispatcher();
+
+                dNotSupported.subscribe(subject, m -> {
+                    coreReceivedSubjectNotSupported.set(m.getSubject());
+                    coreReceivedLatchNotSupported.countDown();
+                });
+
+                ncSupported.createDispatcher().subscribe(subject, m -> {

```

Comment:
Should `ncSupported.createDispatcher()` reuse `dSupported` instead?

Reply:
Yes. fixed.